README.md

```diff
@@ -89,6 +89,32 @@ sudo /usr/local/share/gcm-core/uninstall.sh
 <a name="linux-install-instructions"></a>
 ### Linux
 
+#### Experimental: install from source helper script
+
+If you would like to help dogfood our new install from source helper script,
+run the following:
+
+1. To ensure `curl` is installed:
+
+```shell
+curl --version
+```
+
+If `curl` is not installed, please use your distribution's package manager
+to install it.
+
+0. To download and run the script:
+
+```shell
+curl -o install-from-source.sh -L https://raw.githubusercontent.com/GitCredentialManager/git-credential-manager/main/src/linux/Packaging.Linux/install-from-source.sh &&
+sh ./install-from-source.sh
+git-credential-manager-core configure
+```
```

Comment:
I tried these instructions on my Ubuntu machine, but with this first command:

```
curl -o install-from-source.sh -L https://raw.githubusercontent.com/ldennington/git-credential-manager/ifs-test/src/linux/Packaging.Linux/install-from-source.sh
```

I got these errors at the end:

```
0 upgraded, 0 newly installed, 0 to remove and 32 not upgraded.
Hit:1 http://us.archive.ubuntu.com/ubuntu impish InRelease
Hit:2 http://us.archive.ubuntu.com/ubuntu impish-updates InRelease                                         
Get:3 https://packages.microsoft.com/ubuntu/21.10/prod impish InRelease [10.5 kB]                          
Hit:4 http://us.archive.ubuntu.com/ubuntu impish-backports InRelease             
Hit:5 http://security.ubuntu.com/ubuntu impish-security InRelease
Get:6 https://packages.microsoft.com/ubuntu/21.10/prod impish/main arm64 Packages [567 B]
Get:7 https://packages.microsoft.com/ubuntu/21.10/prod impish/main amd64 Packages [5,298 B]
Get:8 https://packages.microsoft.com/ubuntu/21.10/prod impish/main armhf Packages [567 B]
Fetched 16.9 kB in 0s (34.7 kB/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
32 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package dotnet-sdk-5.0
E: Couldn't find any package by glob 'dotnet-sdk-5.0'
```

I earlier had this output related to the Microsoft feeds:

```
(Reading database ... 195994 files and directories currently installed.)
Preparing to unpack packages-microsoft-prod.deb ...
Unpacking packages-microsoft-prod (1.0-ubuntu21.10.1) over (1.0-ubuntu21.04.1) ...
Setting up packages-microsoft-prod (1.0-ubuntu21.10.1) ...
Installing new version of config file /etc/apt/sources.list.d/microsoft-prod.list ...
```

```ShellSession
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 21.10"
```

Reply:
I updated the script locally to output the URL used for the Microsoft feed: `https://packages.microsoft.com/config/ubuntu/21.10/packages-microsoft-prod.deb` This seems correct to me.

Reply:
Mmm this is very odd. Thanks for pointing it out, will investigate.

Reply:
Super weird - latest test run still green for ubuntu:

https://github.com/ldennington/git-credential-manager/runs/5454412044?check_suite_focus=true

Reply:
yeah, and it's not that an error is being squelched. I see the package being installed correctly.

Reply:
Set up a 21.04 VM locally and it's still running without issue. I also see the following:

```shell
sudo apt install dotnet-sdk-5.0
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
dotnet-sdk-5.0 is already the newest version (5.0.405-1).
0 upgraded, 0 newly installed, 0 to remove and 238 not upgraded.
```

I'm wondering if something went wrong with an `apt update`? Perhaps try running that and then re-running the script?

Reply:
Following up, I found [these instructions](https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#2110-) which use a different download than you suggested, but more importantly has these steps:

```
sudo apt-get update; \
  sudo apt-get install -y apt-transport-https && \
  sudo apt-get update && \
  sudo apt-get install -y dotnet-sdk-5.0
```

I needed these extra steps to make things work (`apt-transport-https`) as an earlier step, before the dotnet install.

Perhaps this will make it work:

```
        # install dotnet packages and dependencies
        $sudo_cmd apt update
        $sudo_cmd apt install apt-transport-https -y
        $sudo_cmd apt update
        $sudo_cmd apt install dotnet-sdk-5.0 dpkg-dev -y
```

(Of course, now that I have the install progressed at this point, my machine becomes less useful as a test bed.)

Reply:
That was also my only issue. The rest of the install succeeded.

Reply:
This was a great find! Thank you so much, Stolee. #1 dogfooder!! 🐶